compiler_opt/rl/compilation_runner.py

```diff
@@ -260,8 +263,8 @@ def collect_data(
       self,
       loaded_module_spec: corpus.LoadedModuleSpec,
       policy: Optional[policy_saver.Policy] = None,
-      reward_stat: Optional[Dict[str, RewardStat]] = None
-  ) -> WorkerFuture[CompilationResult]:
+      reward_stat: Optional[Dict[str, RewardStat]] = None,
+      model_id: Optional[int] = None) -> WorkerFuture[CompilationResult]:
```

Comment:
can you peel `model_id` change out first, make it mandatory, and use it in the name of the temporary policy file in `collect_data` - pass the iteration ID or something from ppo to populate it

reason: right now, if a compilation fails, we can dump out the clang command line (quiet=False - or just log it out, whcih we don't but we could). The input files have a canonical path that makes them easily identifiable back in the corpus. The policy, not so much. This would fix that.

Reply:
Done (#178). The model_id attribute is optional in the CompilationResult because the API for CompilationRunner allows compiling without a policy, but in the data collectors the model_id parameter is required.